In [2]:
import os
from dotenv import load_dotenv

import spotipy
from spotipy.oauth2 import SpotifyClientCredentials

import pandas as pd
import numpy as np
import math
from tqdm import tqdm

In [13]:
artist = pd.read_csv("./processed/selina/artists.csv").drop(["genres"], axis=1)["artist_uri"].to_frame()
print(len(artist), "artists")

18439 artists


In [14]:
tracks = pd.read_csv("./processed/selina/tracks_features.csv")["spotify_track_uri"].to_frame()
print(len(tracks), "tracks")

32644 tracks


In [15]:
artist_track = pd.read_csv("./processed/selina/artist_track.csv")
print(len(artist_track), "artist_track entries")

53086 artist_track entries


In [85]:
missing_tracks = pd.DataFrame(artist_track[~artist_track["track_uri"].isin(all_uris)]["track_uri"].unique(), columns=["track_uri"])
len(missing_tracks)

4406

Looks like 4406 track_uris, that are present in the artist_track dataframe, do not exist in the track dataframe.
This is most likely caused by assigning a random uri from all uris, that point to the same track.

In [74]:
tracks_all_uris = pd.read_csv("./processed/selina/tracks_all_uris.csv")
tracks_all_uris.head(3)

master_metadata_track_name    master_metadata_album_album_name  \
0                  Womanizer             Circus (Deluxe Version)   
1              Boom Boom Pow  THE E.N.D. (THE ENERGY NEVER DIES)   
2          Shut Up And Drive        Good Girl Gone Bad: Reloaded   

                                   spotify_track_uri  \
0           ['spotify:track:4fixebDZAVToLbUCuEloa2']   
1  ['spotify:track:3oDFtOhcN08qeDPAK6MEQG', 'spot...   
2           ['spotify:track:4Tn2llBm1g0UlWctmgPL8Z']   

  master_metadata_album_artist_name  
0                    Britney Spears  
1                   Black Eyed Peas  
2                           Rihanna

In [75]:
tracks_all_uris["spotify_track_uri"] = tracks_all_uris["spotify_track_uri"].replace("\[|\]|\'", "", regex=True).str.split(",")

In [81]:
all_uris = sum(tracks_all_uris["spotify_track_uri"].tolist(), [])

In [84]:
all_uris

['spotify:track:4fixebDZAVToLbUCuEloa2',
 'spotify:track:3oDFtOhcN08qeDPAK6MEQG',
 ' spotify:track:3opVsyWVYEAFK9bJAG8Opa',
 'spotify:track:4Tn2llBm1g0UlWctmgPL8Z',
 'spotify:track:6Q8ljVgg7kR0R5v2aOXHlg',
 ' spotify:track:1FKxKGONukVFXWVJxAKmlz',
 'spotify:track:1ctdfJSzp6qF2iDAVdDrfl',
 'spotify:track:1uVfUdVv0h9MWia3tdZo5G',
 ' spotify:track:29LHe8kG3PraghUZOZYsw4',
 'spotify:track:3E7dfMvvCLUddWissuqMwr',
 ' spotify:track:4TZZvblv2yzLIBk2JwJ6Un',
 'spotify:track:4kLLWz7srcuLKA7Et40PQR',
 'spotify:track:3EDS89HeAhHiZKlljciK0a',
 ' spotify:track:4KktZd9BGHZjW3sK03O4zo',
 'spotify:track:59ZTxQo8rBycIXPLUIKvE2',
 'spotify:track:09mkdGhqb5ySKVsnkx9hy2',
 ' spotify:track:31CsLSgn0HdZEVC8Bnnmvr',
 'spotify:track:1PS1QMdUqOal0ai3Gt7sDQ',
 'spotify:track:455AfCsOhhLPRc68sE01D8',
 ' spotify:track:1htQDV8JxSuXG2QsNj5ttr',
 ' spotify:track:4Smx7N3TVs9McglUSCWEYn',
 ' spotify:track:3oHNJECGN3bBoGXejlw2b1',
 'spotify:track:5mMjkxGijQB4JZallYrkOW',
 ' spotify:track:2ZBNclC5wm4GtiWaeh0DMx',
 'spot